master/internal/core.go

```diff
@@ -1111,6 +1124,10 @@ func buildRM(
 		}
 	}
 
+	if len(clusterNames) != len(rmConfigs) {
+		return nil, fmt.Errorf("resource managers must all have distinct cluster names")
+	}
+
```

Comment:
multiRM name already checks this

Reply:
Checks for distinct naming? If I'm understanding correctly, the previous checks verify that each defined RM has a name, but we don't verify that the names are unique.